qemu/tests/slof_user_interface.py

```diff
@@ -0,0 +1,178 @@
+"""
+slof_user_interface.py include following case:
+ 1. Check the boot menu after press F12 button at the early stage of boot.
+ 2. SLOF user interface testing - boot.
+ 3. SLOF user interface testing - reset-all.
+"""
+import logging
+import os
+import re
+import time
+
+from virttest import error_context
+from provider import slof
+from virttest import data_dir
+from virttest import env_process
+from virttest import utils_misc
+from avocado.utils import process
+from virttest import utils_net
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    Verify SLOF info by user interface.
+
+    Step:
+     Scenario 1:
+      1.1 Boot a guest with at least two blocks, with "-boot menu=on",
+          Press "F12" in the guest desktop at the early stage of booting
+          process.
+      1.2 Check the boot menu info whether are match with guest info.
+      1.3 Select one of valid device to boot up the guest.
+      1.4 Check whether errors in SLOF.
+      1.5 Log in guest successfully.
+      1.6 Ping external host ip successfully.
+
+     Scenario 2:
+      2.1. Boot the guest with spapr-vty and press 's' immediately when
+           the guest boot up.
+      2.2. Check the output of console, whether is stopped enter kernel.
+      2.3. Type "boot" or "reset-all".
+      2.4. Check guest whether boot up successfully.
+      2.5. Log in guest successfully.
+
+    :param test: Qemu test object.
+    :param params: Dictionary with the test parameters.
+    :param env: Dictionary with test environment.
+    """
+    STOP, F12 = range(2)
+    enter_key = {STOP: 's', F12: 'f12'}
+
+    def _create_iso():
+        """ Create iso with one file on it. """
+        if params.get("create_cd", 'no') == 'yes':
+            logging.info("creating iso.")
+            iso_name = params.get("cdrom_cd1")
+            iso = utils_misc.get_path(data_dir.get_data_dir(), iso_name)
+            process.system_output(
```

Comment:
why not call `process.system_output('dd if=/dev/urandom of=new bs=10M count=1&&mkisofs -o %s new...')` once?

Reply:
Thanks, have updated.